avm/utilities/pipelines/staticValidation/psrule/ps-rule.yaml

```diff
@@ -82,3 +82,6 @@ rule:
     - Azure.Storage.UseReplication
     - Azure.AppConfig.PurgeProtect
     - Azure.APIM.MultiRegion # Team agreed this is too expensive for most use cases and is safe to ignore.
+    - Azure.APIM.AvailabilityZone # Only available for Premium SKU
```

Comment:
@jtracey93 what's your take on these suppressions and this one for zones in particular? My last update is we should prioritize WAF alignment (and resiliency in particular) over cost. Is that still the case?

If not, and if we want to go for the suppression, I think we should suppress them only for the defaults test, and keep them active for the waf-aligned one.

Reply:
my vote is to suppress for default tests and leave for WAF. I think most people will not want to deploy a Premium APIM sku by default, and that is unfortunately the only SKU that supports multiple AZs.

Reply:
The `default` and `waf-aligned` test deployments both use Premium SKUs and have multi-az enabled. Since I renamed the test folders for the other SKU deployments so they don't include "default" in the name, maybe we can add these back?

Reply:
Hey, yup this one is a requirement even for the defaults I'm afraid. This is what we state one of the values of AVM modules are, so we must align and make it zone redundant by default and using the premium SKU.

Reply:
Hey @tony-box, as per above comment please consider the following:

- The AVM spec covering commitment for WAF-alignment: https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-sfr2---category-composition---waf-aligned
- There we mention WAF **should** be set by default whenever possible. That should be prioritized over cost optimization, but it **must** be possible to override the values if needed.
- This seems very applicable to the premium sku for APIM. It is Premium by default to allow WAF properties available only for Premium. However, different skus are possible by overriding the default value. This is what the test cases for different skus are showing.

Therefore, please remove the 4 APIM exceptions from the general PSRule settings and:
- if possible, values making PSRule compliant should be set as default values
- if not possible, e.g. if they involve recommendations requiring an external dependency/resource to be deployed and configured, then we still need to try having the waf-aligned test compliant. Default tests in these exceptional cases can have suppressions. The right file where to add them is not the general PSRule settings, but this one: https://github.com/Azure/bicep-registry-modules/blob/main/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/min-suppress.Rule.yaml

Reply:
@eriqua -- already removed the exceptions in my modified PR and have been trying to get it to pass validation tests. Am  having trouble with the WAF test and some (seemingly) unrelated permission/identity issue.